test/built-ins/TypedArray/prototype/slice/results-with-empty-length.js

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.slice
+description: slice may return a new empty instance
+includes: [testTypedArray.js, compareArray.js]
+---*/
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([40, 41, 42, 43]);
+
+  function testRes(result, msg) {
+    assert.sameValue(result.length, 0, msg);
+    assert.sameValue(
+      result.hasOwnProperty(0),
+      false,
+      msg + " & result.hasOwnProperty(0) === false"
+    );
+  }
+
+  testRes(sample.slice(4), "begin == length");
+  testRes(sample.slice(5), "begin > length");
+
+  testRes(sample.slice(4, 4), "begin == length, end == length");
+  testRes(sample.slice(5, 4), "begin > length, end == length");
+
+  testRes(sample.slice(4, 4), "begin == length, end > length");
+  testRes(sample.slice(5, 4), "begin > length, end > length");
```

Comment:
It looks like you meant to specify `5` as the second argument in these two calls (otherwise, the assertions are identical to the two that precede them).